task/cache.py

```diff
@@ -55,13 +56,13 @@ def prune(self):
         if not os.path.exists(self.directory):
             return
         now = time.time()
-        try:
-            for filename in os.listdir(self.directory):
-                path = os.path.join(self.directory, filename)
-                if os.path.isfile(path) and os.stat(path).st_mtime < now - 7 * 86400:
+        for filename in os.listdir(self.directory):
```

Comment:
Missed this before.

This is screaming out for porting to `scandir`, which would let you avoid the pathname computations and also the stat call for `os.path.isfile()`.  Note also the `.stat()` method on the returned object (but no `.unlink()`).

Reply:
Done in #4051, I'll drop this commit.